fix: FIT-531: Fix flickering when switching from uploading to uploaded on Import modal

ricardoantoniocm opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/17097963844)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/520
<!-- Follow Merge -->










## Reason for change

When uploading files manually the transition between the uploading state and the uploaded state is not smooth. The empty state is shown briefly.

This PR addresses that issue and includes improvements for the layout, and animations to highlight recently uploaded files.

This also implements truncation for the file names so that they don't extend beyond a single line, ensuring a natural visual flow of each file upload.

In addition to the above, the file size of uploaded files is now shown (does not work currently on LSE). 

## Before
https://www.loom.com/share/f264a81ad8b74a8e9f780fc745c8b597?sid=c4c469fe-6a3e-436f-9841-04f27d84b9a1

## After
https://www.loom.com/share/0eef61547f304c1ebe89164b4a72eb51?sid=2bb3de0a-6a06-47cb-bb95-d6355447f49d


<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

ricardoantoniocm committed:
fix: FIT-531: Import step flickers when switching from uploading to completed state

ricardoantoniocm committed:
Fixes glitch when changing from uploading to uploaded state. Improves order of uploaded files to prevent visual glitches and adds animation to highlight recently uploaded files. Implements helper to format file size. Fixed the issue that file sizes weren't being displayed.

ricardoantoniocm committed:
Fixes animation when file is uploaded.

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16991081310)
> Successfully merged:  create mode 100644 web/libs/editor/tests/integration/e2e/control_tags/text-with-dual-taxonomy.cy.ts

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-531/import-step-flickers-when-switching-uploading-completed-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16991081310

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17070255672

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-fit-531/import-step-flickers-when-switching-uploading-completed-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17070255672

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17070495485

jombooth commented on label_studio/data_import/serializers.py:
```diff
@@ -21,7 +21,14 @@ class Meta:
-        fields = ['id', 'file']
+        fields = ['id', 'file', 'size']
+
+    def get_size(self, obj):
```
```suggestion
    def get_size(self, obj) -> int | None:
```

This will help Spectacular accurately type the field in responses; otherwise it will just be a string

ricardoantoniocm committed:
Apply suggestion from @jombooth

Co-authored-by: Jo Booth <jo.m.booth@gmail.com>

ricardoantoniocm committed:
Truncate name with css when the screen is too small.

ricardoantoniocm committed:
Merge branch 'fb-fit-531/import-step-flickers-when-switching-uploading-completed-state' of https://github.com/heartexlabs/label-studio into fb-fit-531/import-step-flickers-when-switching-uploading-completed-state

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17078851950

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-fit-531/import-step-flickers-when-switching-uploading-completed-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17078851950

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17079714393

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17079791124

jombooth committed:
empty commit to trigger SDK rebuild

jombooth committed:
Merge branch 'fb-fit-531/import-step-flickers-when-switching-uploading-completed-state' of github.com:heartexlabs/label-studio into fb-fit-531/import-step-flickers-when-switching-uploading-completed-state

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17079858115

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17079955763

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17080391434

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17080467980

ricardoantoniocm commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17086845160)
> Successfully merged:

ricardoantoniocm committed:
Merge branch 'develop' into 'fb-fit-531/import-step-flickers-when-switching-uploading-completed-state'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17086845160

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17086950523

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17097963844

ricardoantoniocm committed:
Empty commit to trigger CI/CD or mark a milestone
